Modules/_threadmodule.c

```diff
@@ -1159,6 +1159,28 @@ allocated consecutive numbers starting at 1, this behavior should not\n\
 be relied upon, and the number should be seen purely as a magic cookie.\n\
 A thread's identity may be reused for another thread after it exits.");
 
+static PyObject *
+thread_get_tid(PyObject *self, PyObject *Py_UNUSED(ignored))
+{
+    unsigned long tid = PyThread_get_thread_id();
+    return PyLong_FromUnsignedLong(tid);
+}
+
+PyDoc_STRVAR(get_tid_doc,
+"get_tid() -> integer\n\
+\n\
+Return a non-negative integer identifying the thread as reported\n\
+by the OS (kernel). This may be used to uniquely identify a\n\
+particular thread within a system.\n\
+\n\
+The behavior of this function varies between operating systems:\n\
```

Comment:
Usually the detailed documentation would go into `threading.rst`, and the docstring is terser.